coalib/settings/Setting.py

```diff
@@ -144,6 +149,29 @@ def __path_list__(self):
         """
         return [Setting.__path__(elem, self.origin) for elem in self]
 
+    def __url__(self):
+        """
+        Determines the url validity of this setting.
+
+        :return: url string
+        :raises ValueError: If the url is not valid.
+        """
+        domain1_re = r'(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)'
+        domain2_re = r'(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)'
+        regex = re.compile(
+            r'^(?:http|ftp)s?://'
+            r'(?:' + domain1_re + '+' + domain2_re + '|'
+            r'localhost|'
+            r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'
+            r'(?::\d+)?'
+            r'(?:/?|[/?]\S+)$', re.IGNORECASE)
```

Comment:
@sils1297 From what it is mentioned in the stackoverflow link.It is from django 1.3
core/validators.py line47
the license of django is https://github.com/django/django/blob/master/LICENSE